Source/SwiftLintFramework/Rules/ExplictSelfRule.swift

```diff
@@ -0,0 +1,286 @@
+        "\(type) Bad4 { func value() -> Int { return 42 }; "
+            + "func function() -> Int { return value() } }",
+        "\(type) Bad4a { func value() -> Int { return 42 }; "
+            + "func function() -> Int { return value() } }",
+        "\(type) Bad4b { func value(foo: Int) -> Int { return foo }; "
+            + "func function() -> Int { return value(42) } }",
+        "\(type) Bad4c { func value(foo: Int, bar: Int) -> Int { return foo + bar }; "
+            + "func function() -> Int { return value(40, bar: 2) } }",
+        "\(type) Bad4d { func value(foo bar: Int, baz: Int) -> Int { return bar + baz }; "
+            + "func function() -> Int { return value(foo: 40, baz: 2) } }",
+        "\(type) Bad5 { func value() -> Int { return 42 }; var property: Int { return value() } }",
+        "\(type) Bad6 { func value() { }; init() { value() } }",
+        "\(type) Bad7 { let value: Int = 42 }; "
+            + "extension Bad7 { var property: Int { return value } }",
+        "\(type) Bad8 { func value() -> Int { return 42 } }; "
+            + "extension Bad8 { var property: Int { return value() } }",
+        "\(type) Bad9 { let value: Int = 42 }; "
+            + "extension Bad9 { func function() -> Int { return value } }",
+        "\(type) Bad10 { func value() -> Int { return 42 } }; "
+            + "extension Bad10 { func function() -> Int { return value() } }",
+        "\(type) Bad11 { static var value: Int = 42; static var property: Int { return value } }",
+        "\(type) Bad12 { static func value() -> Int { return 42; }; "
+            + "static var property: Int { return value() } }",
+        "\(type) Bad13 { static var value: Int = 42; "
+            + "static func function() -> Int { return value } }",
+        "\(type) Bad14 { static func value() -> Int { return 42; }; "
+            + "static func function() -> Int { return value() } }"
+    ]
+}
+let triggeringClassExamples = [
+    "class Bad15_A { var value: Int = 42 }; "
+        + "class Bad15_B: Bad15_A { var property: Int { return value } }",
+    "class Bad16_A { func value() -> Int { return 42 } }; "
+        + "class Bad16_B: Bad16_A { var property: Int { return value() } }",
+    "class Bad17_A { var value: Int = 42 }; "
+        + "class Bad17_B: Bad17_A { func function() -> Int { return value } }",
+    "class Bad18_A { func value() -> Int { return 42 } }; "
+        + "class Bad18_B: Bad18_A { func function() -> Int { return value() } }",
+]
+
+let nonTriggeringExamples = ["class", "struct"].flatMap(nonTriggering)
+let triggeringExamples = ["class", "struct"].flatMap(triggering)
+
+public struct ExplicitSelfRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Error)
+
+    public init() { }
+
+    public static let description = RuleDescription(
+        identifier: "explicitSelf",
+        name: "Explict Self",
+        description: "Require explicit self for instance members",
```

Comment:
The wording seems weird. How about "Explicit `self` for instance members and functions is required".